libs/test-utils/src/system/time_travel.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General
+
+use sp_runtime::traits::{CheckedAdd, Header, One};
+
+/// Advances the chain `n` number of blocks
+/// for use with tests where actions would need to take
+/// place a certain number of blocks after another action.
+pub fn advance_n_blocks<T: frame_system::Config>(n: u64) {
+	for _ in 0..n {
+		let h = frame_system::Pallet::<T>::finalize();
+		let b = h
+			.number()
+			.checked_add(&T::BlockNumber::one())
+			.expect("Mock block count increase failed");
+		frame_system::Pallet::<T>::initialize(&b, h.parent_hash(), h.digest());
+	}
+}
```

Comment:
If you only need to advance blocks, you can use `frame_system::Pallet<T>::set_block_number()` utility under `std` and `runtime-benchmarks`

Reply:
That is true. However, if any hook from `on_initialize` or `on_finalize` is required, the above approach should be taken instead.

Reply:
Agree

Reply:
For this case it could be done with setting the block number.  My preference would be to keep this as is as it is more generic, and how the chain would actually function, but can also change.

Reply:
No strong opinion 👍🏻

Only highlight that in order to leave this generically usable, we would need to call `on_finalize() / on_initialize()` over all runtime pallets. i.e. `pallet_timestamp` also change things between blocks, that could be required for the user of this function.

Reply:
It is really hard to generate a correct block -- i.e. run all hooks of a runtimes pallet. This is the main reason I build Fudge. But feel free to try it. For now, I think we can keep it, although, I agree with Luis.